sopel_remind/backend.py

```diff
@@ -136,24 +145,78 @@ def parse_in_delta(line: str) -> Tuple[timedelta, str]:
     return delta, message
 
 
-def parse_at_time(line: str) -> Tuple[time, str]:
+def parse_at_time(
+    line: str,
+    today: datetime,
+    user_tz: pytz.BaseTzInfo,
+) -> Tuple[datetime, str]:
     """Parse a reminder line using the ``at`` command format.
 
-    :param line: reminder line from the ``in`` command
-    :return: a 2-value tuple with ``(seconds, message)``
-    :raise ValueError: when ``line`` doesn't match the ``in`` command format
+    :param line: reminder line from the ``at`` command
+    :param today: today (localized in the user's timezone)
+    :param user_tz: user's timezone
+    :return: a 2-value tuple with ``(when, message)``
+    :raise ValueError: when ``line`` doesn't match the ``at`` command format
+
+    The goal is to parse the line to get what to remind the user and when
+    exactly. The format can be one of the following:
+
+    * a date only
+    * a time only
+    * a date before the time
+    * a time before the date
+
+    These formats requires some manipulation of the current time (``today``)
+    in the appropriate timezone (``tz``) to always return an accurate
+    :class:`datetime.datetime` object.
     """
     result = AT_RE.match(line)
 
     if not result:
         raise ValueError('Invalid at arguments: %r' % line)
 
-    hours = int(result.group('hours') or 0)
-    minutes = int(result.group('minutes') or 0)
-    seconds = int(result.group('seconds') or 0)
+    groups = result.groupdict()
+    time_only = groups['t']
+    date_only = groups['d']
+    date_time = groups['d_t']
+    time_date = groups['t_d']
+
+    raw_date: str = today.strftime('%Y-%m-%d')
+    raw_time: str = today.strftime('%H:%M:%S')
+
+    if time_only:
+        raw_time = str(time_only)
+    elif date_only:
+        raw_date = str(date_only)
+    elif date_time:
+        raw_date, raw_time = date_time.split(' ')
+    else:
+        raw_time, raw_date = time_date.split(' ')
```

Comment:
I don't think it's possible to reach this point in the code if none of the four cases in the regex matches, but this bare `else` still rubs me wrong because trying to do `None.split(' ')` is catastrophic. 🙂

My habit is to do (admittedly silly) things like:

```suggestion
    if time_only:
        raw_time = str(time_only)
    elif date_only:
        raw_date = str(date_only)
    elif date_time:
        raw_date, raw_time = date_time.split(' ')
    elif time_date:
        raw_time, raw_date = time_date.split(' ')
    else:
        raise RuntimeError("No regex group captured anything. This shouldn't be possible.")
```

Reply:
Yeah, I did that at first, but that means I can't reach that case, so I lose coverage. And I don't want that. :grin:

Reply:
And yes, it is not possible to reach that stage if the regex matches.